paperslicer/metadata/providers/crossref.py

```
@@ -0,0 +1,78 @@
+    def __init__(self, mailto: Optional[str] = None, timeout: int = 10):
+        self.base = "https://api.crossref.org"
+        self.timeout = timeout
+        mail = mailto or os.getenv("CROSSREF_MAILTO") or "you@example.com"
+        self.headers = {"User-Agent": f"PaperSlicer/0.1 (mailto:{mail})"}
+
+    def by_doi(self, doi: str) -> Optional[Dict[str, Any]]:
+        try:
+            r = requests.get(f"{self.base}/works/{doi}", headers=self.headers, timeout=self.timeout)
+            if r.status_code == 200:
+                return r.json().get("message", {})
+        except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)